File: leetcode-151-Reverse_Words_in_a_String/leetcode-151-Reverse_Words_in_a_String.cpp
#include <iostream>
#include <cstdlib>
#include <string>


std::string reverseWords(std::string s) {
    std::string res;
    size_t i = 0;
    while(i<s.size()){
        while(i<s.size() && s[i] == ' '){
            ++i;
        }
        std::string temp_s;
        while(i<s.size() && s[i] != ' '){
            temp_s += s[i];
            ++i;
        }
        if(temp_s.empty()){
            break;
        }
        if(!res.empty()){
            temp_s+=' ';
        }
        res = temp_s+res;
    }
    return res;
}


int main(){
    std::string s = "the sky is blue";

    auto res = reverseWords(s);
    
    std::cout << res << std::endl;

    return EXIT_SUCCESS;
}

File: leetcode-151-Reverse_Words_in_a_String/leetcode-151-Reverse_Words_in_a_String.py
class Solution:
    def reverseWords(self, s: str) -> str:
        words_lst: list[str] = s.strip().split(" ")
        words_lst.reverse()
        words_lst = [c for c in words_lst if c != ""]
        return " ".join(words_lst)


if __name__ == "__main__":
    s: str = "the sky is blue"
    result = Solution().reverseWords(s)
    print(result)
